# throw

Throws a developer-specified exception, which can be caught
 with a catch block.

```javascript
throw([message] [, type] [, detail] [, errorcode] [, extendedinfo] [, object])
```

```javascript
returns numeric
```

## Argument Reference

| Name | Type | Required | Default | Description |
| --- | --- | --- | --- | --- |
| message | string | No |  | Message that describes exception event. |
| type | string | No | Custom | * A custom type<br /> * Application<br /> Do not enter another predefined type; types are not<br /> generated by CFML applications. If you specify<br /> Application, you need not specify a type for cfcatch. |
| detail | string | No |  | Description of the event. CFML appends error position<br /> to description; server uses this parameter if an error is<br /> not caught by your code. |
| errorcode | string | No |  | A custom error code that you supply. |
| extendedinfo | string | No |  | Additional custom error data that you supply. |
| object | any | No |  | Requires the value of the cfobject tag name attribute.<br /><br /> Throws a Java exception from a CFML tag.<br /><br /> This attribute is mutually exclusive with all other<br /> attributes of this tag. |

## Throw a custom exception

Use the throw function to throw a custom application exception.

```javascript
<cfscript>
throw(type="MyCustomError", message="A custom error has been thrown!");
</cfscript>
```

## Throw a custom http response exception

Use the throw function to throw a custom exception when the http response is invalid.

```javascript
if (!isJSON(httpResponse.fileContent)) {
	throw(type="InvalidHTTPResponse",message="The http response was not valid JSON");
}
```
